packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -110,13 +112,52 @@ export class ProjectService {
     }: ProjectServiceDependencies) {
         this.projectModel = projectModel;
         this.onboardingModel = onboardingModel;
-        this.projectAdapters = {};
+        this.warehouseClients = {};
         this.savedChartModel = savedChartModel;
         this.jobModel = jobModel;
         this.emailClient = emailClient;
         this.spaceModel = spaceModel;
     }
 
+    private async _getWarehouseClient(
+        projectUuid: string,
+    ): Promise<WarehouseClient> {
+        // Always load the latest credentials from the database
+        const credentials =
+            await this.projectModel.getWarehouseCredentialsForProject(
+                projectUuid,
+            );
+        // These clients reconnect for every query so no benefit to caching
```

Comment:
If there is no difference can't we cache them anyway and have the same logic for all warehouses?

Reply:
Good suggestion!